File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int min = INT_MAX;
        int r = nums.size() - 1;
        while(l <= r)
        {
            if (nums[l] < min)
                min = nums[l];
            if (nums[r] < min)
                min = nums[r];
            l++;
            r--;
        }
        return min;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.py
from heapq import heappop, heappush, heapify
class Solution:
    def findKthLargest(self, nums: list[int], k: int) -> int:
        # min heap
        # store up to k elements on the heap and return the first element as a res
        n = len(nums)
        heap = [num for num in nums[:k]]
        heapify(heap)
        for num in nums[k:]:
            if num > heap[0]:
                heappop(heap)
                heappush(heap, num)
        return heap[0]